src/CommunityToolkit/Xamarin.CommunityToolkit/Views/HexLayout.shared.cs

```diff
@@ -0,0 +1,278 @@
+﻿using System;
+using Xamarin.Forms;
+
+namespace Xamarin.CommunityToolkit.UI.Views
+{
+    /// <summary>
+    /// A Layout that arranges the elements in a honeycomb pattern.
+    /// Based on https://github.com/AlexanderSharykin/HexGrid. Created by Alexander Sharykin.
+    /// </summary>
+    public class HexLayout : Layout<View>
+    {
+        public static readonly BindableProperty OrientationProperty =
+         BindableProperty.Create(nameof(Orientation), typeof(StackOrientation), typeof(HexLayout), StackOrientation.Vertical,
+             BindingMode.TwoWay, propertyChanged: (bindable, oldvalue, newvalue) => ((HexLayout)bindable).InvalidateMeasure());
+
+        public StackOrientation Orientation
```

Comment:
I wonder whether we should expose our own enum HexOrientation?

Reply:
Yes, added `HexOrientation`.